### ISSUE TITLE
Error while chef-ing worker node.

### ISSUE THREAD

**Issue by amithkanand** [bloomberg/chef-bach#291]
After introducing `chef-vault`, cluster build currently throws an error while trying to create the `zabbix-agentd.conf` file. Below is the error message
```` 
================================================================================
Error executing action `create` on resource 'template[/usr/local/etc/zabbix_agentd.conf]'
================================================================================

Chef::Mixin::Template::TemplateError
------------------------------------
os/mysql-zabbix is not encrypted with your public key.  Contact an administrator of the vault item to encrypt for you!
```

**Comment by DrRayWang** [comment]
The reason is :
zabbix-head.rb will chef vault mysql-zabbix-password with chef vault resource.This recipe only runs on Head nodes.
But zabbix-worker.rb run template of this file : cookbooks/bcpc/templates/default/zabbix_agentd.conf.erb, which needs access to mysql-zabbix-password using chef vault ruby API. find_resource can only find chef vault resource, not chef vault ruby API. This results in execution failure for worker nodes when it run zabbix-worker.rb, and needs access to chef-vaulted mysql-zabbix-password.

**Comment by kiiranh** [comment]
Fixed by #292 . Closing
